_posts/2023-10-02-intellij-quarkus-recap.adoc

```diff
@@ -0,0 +1,178 @@
+---
+layout: post
+title: A recap of Quarkus Tools for IntelliJ's latest improvements
+date: 2023-10-02
+tags: intellij ide
+synopsis: Over the past five months, we’ve been really hard at work to improve Quarkus Tools in every possible way. So let’s look in more details at what was accomplished.
+author: fbricon
+---
+:imagesdir: /assets/images/posts/intellij-quarkus-tools-oct23
+
+https://plugins.jetbrains.com/plugin/13234-quarkus-tools[Quarkus Tools for IntelliJ] is a free and open source extension, helping users develop Quarkus applications by providing content-assist, validation, run configurations and many other features right from their favourite IDE.
+
+This extension is based on the https://github.com/eclipse/lsp4mp[LSP4MP] (i.e. MicroProfile) and its https://github.com/redhat-developer/quarkus-ls/tree/master/quarkus.ls.ext[Quarkus] add-on, and the https://github.com/redhat-developer/quarkus-ls/tree/master/qute.ls[Qute language server]. These are all used in our popular  https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus[Quarkus Tools extension] for Visual Studio Code.
+
+Over the past five months, we've been *really* hard at work to improve Quarkus Tools in every possible way, averaging about one release every two weeks.
+
+We've been focusing on performance improvements, Quarkus 3.x with JakartaEE namespace support, Qute templating support, and generally trying to keep Quarkus Tools out of your way. So let's look in more details at what was accomplished.
+
+== Increased stability and performance
+We've addressed various critical issues in our custom Language Server client (LSP4IJ), that previously caused Language Servers to freeze IntelliJ IDEA. Improvements include:
+
+- Waiting until indexing is finished before launching the language servers.
+- Language server requests no longer blocking the editor.
+- Debouncing some requests to minimize heavy processing
+- New request cancellation support to avoid unnecessary processing.
+- Faster validation and lazy loading of quick fixes.
+- Pre-emptively fetching Quarkus data in the Quarkus project wizard, in order to avoid waiting after clicking on the `Next` button.
```

Comment:
A big improvement too is that we implemented LSP resolve support for completion and code actions. 

 * for instance for Quarkus properties completion, before the JSON response stored the documentation for each properties which is a so big JSON content. Today we receive all properties without documentation and we get the documentation only for a property which is selected (to show teh documentation in the second popup).
 * for code action we resolve the code actiononly when user click on the quickfix. Before we resolved all quickfixes for all diagnostics. As we have Java quickfix which uses AST, those compute can be very expensive, and more if we need to compute all code actions for all diagnostics.

Reply:
added bit about reducing completion payload size.
Lazy loading of quickfixes was already mentioned on the above line